cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/dataset/MultipleOutputs.java

```diff
@@ -0,0 +1,280 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.runtime.batch.dataset;
+
+import co.cask.cdap.common.lang.Instantiator;
+import co.cask.cdap.common.lang.InstantiatorFactory;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.reflect.TypeToken;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.mapreduce.Counter;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.OutputFormat;
+import org.apache.hadoop.mapreduce.RecordWriter;
+import org.apache.hadoop.mapreduce.StatusReporter;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.apache.hadoop.mapreduce.TaskInputOutputContext;
+import org.apache.hadoop.mapreduce.task.TaskAttemptContextImpl;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.StringTokenizer;
+
+/**
+ * The MultipleOutputs class simplifies writing output data to multiple outputs.
+ * It has been adapted from org.apache.hadoop.mapreduce.lib.output.MultipleOutputs.
+ */
+public class MultipleOutputs {
+
+  private static final String MULTIPLE_OUTPUTS = "hconf.mapreduce.multipleoutputs";
+
+  private static final String MO_PREFIX = MULTIPLE_OUTPUTS + ".namedOutput.";
+
+  private static final String FORMAT = ".format";
+  private static final String KEY = ".key";
+  private static final String VALUE = ".value";
+  private static final String CONF = ".conf";
+
+  /**
+   * Cache for the taskContexts
+   */
+  private Map<String, TaskAttemptContext> taskContexts = new HashMap<>();
+
+  /**
+   * Checks the existence of a named output within a collection.
+   *
+   * @throws IllegalArgumentException if the output name is not existing/absent, as appropriate.
+   */
+  private static void checkNamedOutputName(String namedOutput, Collection namedOutputs, boolean expectToExist) {
```

Comment:
Why the `Colleciton` is without type parameter? Don't you see a warning in IntelliJ?